packages/apps/job-launcher/server/src/modules/manifest/manifest.service.spec.ts

```diff
@@ -0,0 +1,851 @@
+            type: mockRequestType,
+            user_guide: mockDto.userGuide,
+            segment_duration: mockDto.segmentDuration,
+          },
+          data: {
+            data_url: generateBucketUrl(mockDto.data.dataset, mockRequestType)
+              .href,
+          },
+          job_bounty: ethers.formatEther(jobBounty),
+          validation: {
+            gt_url: generateBucketUrl(mockDto.groundTruth, mockRequestType)
+              .href,
+            min_quality: mockDto.minQuality,
+          },
+        });
+      });
+    });
+
+    describe('createHCaptchaManifest', () => {
+      const requestType = HCaptchaJobType.HCAPTCHA;
+      const tokenFundAmount = faker.number.int({ min: 1, max: 1000 });
+      const tokenFundDecimals = faker.number.int({ min: 1, max: 18 });
+
+      beforeEach(() => {
+        const fileContent = JSON.stringify({
+          [faker.internet.url()]: [true, true, true],
+        });
+        (listObjectsInBucket as jest.Mock).mockResolvedValueOnce([
+          `${faker.word.sample()}.jpg`,
+          `${faker.word.sample()}.jpg`,
+          `${faker.word.sample()}.jpg`,
+        ]);
+        mockStorageService.uploadJsonLikeData.mockResolvedValueOnce({
+          url: faker.internet.url(),
+          hash: faker.string.uuid(),
+        });
+        mockStorageService.downloadJsonLikeData.mockResolvedValueOnce(
+          fileContent,
+        );
+      });
+
+      it('should create a valid HCaptcha manifest for COMPARISON job type', async () => {
+        const jobDto = createJobCaptchaDto({
+          annotations: {
+            typeOfJob: JobCaptchaShapeType.COMPARISON,
+            labelingPrompt: faker.lorem.sentence(),
+            groundTruths: faker.internet.url(),
+            taskBidPrice: faker.number.float({ min: 0.1, max: 10 }),
+            exampleImages: [faker.internet.url(), faker.internet.url()],
+          },
+        });
+
+        const result = await manifestService.createManifest(
+          jobDto,
+          requestType,
+          tokenFundAmount,
+          tokenFundDecimals,
+        );
+
+        expect(result).toEqual({
+          job_mode: JobCaptchaMode.BATCH,
+          requester_accuracy_target: jobDto.accuracyTarget,
+          request_config: expect.any(Object),
+          restricted_audience: expect.any(Object),
+          requester_max_repeats: jobDto.maxRequests,
+          requester_min_repeats: jobDto.minRequests,
+          requester_question: { en: jobDto.annotations.labelingPrompt },
+          job_total_tasks: expect.any(Number),
+          task_bid_price: jobDto.annotations.taskBidPrice,
+          taskdata_uri: expect.any(String),
+          public_results: true,
+          oracle_stake: expect.any(Number),
+          repo_uri: expect.any(String),
+          ro_uri: expect.any(String),
+          request_type: expect.any(String),
+          groundtruth_uri: expect.any(String),
+          requester_restricted_answer_set: expect.any(Object),
```

Comment:
Might we want to check for exact values here and in other similar places or is `expect.any` enough?